### PR TITLE
Implement `Collection.Compact` for SQLite

### LINKED ISSUE #3469: Implement `Collection.Compact` for SQLite
### What should be done?

SQLite's backend `Collection.Compact` method should call [`PRAGMA incremental_vacuum`](https://www.sqlite.org/pragma.html#pragma_incremental_vacuum) and then [`ANALYZE`](https://www.sqlite.org/lang_analyze.html) on the given collection/table.

When `Full` is true, [`VACUUM`](https://www.sqlite.org/lang_vacuum.html) should be run instead of `PRAGMA incremental_vacuum`.

If the database or collection does not exist, `ErrorCodeDatabaseDoesNotExist` or `ErrorCodeCollectionDoesNotExist` should be returned (that's already checked by the contact and backend test).

### Where?

* https://github.com/FerretDB/FerretDB/blob/main/internal/backends/sqlite/collection.go
* https://github.com/FerretDB/FerretDB/blob/main/internal/backends/collection_test.go

### Definition of Done

- SQLite backend updated;
- `Skip` removed from the test.

### PR DESCRIPTION
**Pull request by Akhil-2001**
# Description

Closes #3469.

## Readiness checklist

- [x] I added/updated unit tests (and they pass).
- [ ] I added/updated integration/compatibility tests (and they pass).
- [ ] I added/updated comments and checked rendering.
- [ ] I made spot refactorings.
- [ ] I updated user documentation.
- [x] I ran `task all`, and it passed.
- [x] I ensured that PR title is good enough for the changelog.
- [ ] (for maintainers only) I set Reviewers ([`@FerretDB/core`](https://github.com/orgs/FerretDB/teams/core)), Milestone (`Next`), Labels, Project and project's Sprint fields.
- [x] I marked all done items in this checklist.

### CODE REVIEW

File: internal/backends/sqlite/collection.go [hunk 1]
```diff
@@ -322,7 +322,39 @@ func (c *collection) Stats(ctx context.Context, params *backends.CollectionStats
...
+	if _, err = db.ExecContext(ctx, q); err != nil {
+		return nil, lazyerrors.Error(err)
+	}
+
+	// Call ANALYZE to update statistics of tables and indexes,
+	// see https://www.sqlite.org/lang_analyze.html.
+	q = `ANALYZE`
+	if _, err = db.ExecContext(ctx, q); err != nil {
+		return nil, lazyerrors.Error(err)
+	}
```
Comment (AlekSi) [inline internal/backends/sqlite/collection.go:356]: No, that should not be done in this method

**Comment by AlekSi** [comment]
@Akhil-2001 please click the CLA button in this comment: https://github.com/FerretDB/FerretDB/pull/3533#issuecomment-1753183081

File: internal/backends/sqlite/collection.go (hunk 1 above)
Comment (Akhil-2001) [reply internal/backends/sqlite/collection.go:356]: Right, let me know where `ANALYZE` should be called.

**Comment by Akhil-2001** [comment]
Turns out CLA is failing due to a wrong email address configured with my commits. Closing this PR to raise a new one.

**Pull request closed without merging**